### DISCUSSION TITLE
How can I avoid errors when using equations in German text?

### DISCUSSION THREAD

**Discussion by StrawberryOwl** [valentjn/vscode-ltex#276] [Q&A] [answered]
I am writing a quite long document using a lot of mathematical equations, mostly inside equation and align environments. The problem is that I always get the error "Außer am Satzanfang werden nur Nomen und Eigennamen großgeschrieben. – DE_CASE" (I am writing the document in German) when I start a sentence after an equation. So it wants me to start the new line with a small letter because it sees the equation as part of a sentence which I didn't finish with a dot. But the equation is no sentence (I have a sentence above ending with a dot) and I want to start a new sentence.
I already tried using the ignore setting (ltex.latex.environments and there set environment, align and environment* and align* to "ignore") but this doesn't change anything. When I end the equation with a dot, the error disappears. This doesn't seem right to me and I do not want to hide them all as false positives (as it appears to be a systematic error).
Am I using the settings wrong or this a bug?

**Comment by valentjn** [comment, marked as answer]
I need an example for the alleged false positive diagnostic. Otherwise, I'd say this is desired behavior of LT<sub>E</sub>X.

According to DIN 1338 *Formelschreibweise und Formelsatz* (see 4.7.2 in this [excerpt](https://tu-dresden.de/ing/elektrotechnik/ifa/ressourcen/dateien/richtlinie_wiss_arbeiten/auszug_din_1338.pdf)) and most typographically good examples from literature, equations and formulas, even and especially when &ldquo;displayed&rdquo; (i.e., not inline), are in most cases to be treated as part of a sentence, except in some special cases, for example, if formulas are listed in a table. Just a formula on its own is not a sentence, just like a very long number on its own isn't, but a text body (i.e., your document minus headings, floats, headers/footers, etc.) should only consist of sentences. This means you can't just put a formula in your text body, but you should incorporate it into a sentence, e.g., by ending the previous sentence with a colon (just like this sentence now):

```
Der Beweis basiert auf folgender langer Rechnung:
\begin{equation}
  E = mc^2.
\end{equation}
Hier haben wir die Äquivalenz von Energie und Masse verwendet.
```

This is also better from a stylistic/semantic point of view, as you shouldn't just dump stuff like an equation into the text body, but interact with it, i.e., reference it (like `folgender Rechnung`) and explain its purpose (like in the second sentence).

Regarding [`ltex.latex.environments`](https://valentjn.github.io/vscode-ltex/docs/settings.html#ltexlatexenvironments), the docs say that this setting only affects environments that are not already handled by default, but equations environments are. As a workaround, you can either define your own alias environments for `align` etc. via `\newenvironment` and add them to `ltex.latex.environments`, or you can use [magic comments](https://valentjn.github.io/vscode-ltex/docs/advanced-usage.html#magic-comments), which would be more tedious of course. But I wouldn't recommend either of the workarounds, for the reasons I've explained above.

**Reply by StrawberryOwl** [reply]
Thank you for your very detailed answer! I didn't even know that there was a DIN norm for this (although I probably should have suspected).
Then I will just get used to writing a dot after my equations and enjoy using your amazing spell checker even more!
